### PR TITLE
mgr/cephadm: fix 'cephadm osd activate' on existing osd devices

### PR DESCRIPTION
**Pull request by liewegas**
## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/services/osd.py [hunk 1]
```diff
@@ -125,6 +125,10 @@ async def deploy_osd_daemons_for_existing_osds(self, host: str, service_name: st
                 if osd_id in before_osd_uuid_map and osd_id not in replace_osd_ids:
                     # if it exists but is part of the replacement operation, don't skip
                     continue
+                if self.mgr.cache.has_daemon(f'osd.{osd_id}'):
+                    # cephadm daemon instance already exists
+                    logger.debug(f'osd id {osd_id} daemon already exists')
+                    continue
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/services/osd.py:131]: If you physically move a disk to a different host, you end up here, right? 

https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/serve.py#L446-L472

**Review by sebastian-philipp — APPROVED** [review]

**Review by sebastian-philipp — APPROVED** [review]

**Comment by liewegas** [comment]
jenkins test api

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-12-16_13:33:14-orch:cephadm-wip-swagner3-testing-2021-12-15-1525-distro-default-smithi/

**Comment by liewegas** [comment]
https://pulpito.ceph.com/sage-2021-12-16_22:01:08-orch:cephadm-wip-sage2-testing-2021-12-16-1045-distro-basic-smithi/

**Pull request merged**